src/gluonts/dataset/common.py

```diff
@@ -145,6 +170,44 @@ class FileDataset(Dataset):
         Indicates whether the dataset should be cached or not.
     """
 
+    def __new__(  # type: ignore
```

Comment:
Defining `__new__` seems a little odd. Does this need to be a class now? Could it be a function, or an alternative constructor for `DatasetCollection` (something like `from_files`)?

For example I'm not sure when `FileDataset.__iter__` is invoked with this change